### DISCUSSION TITLE
Define environnement variables at run time (client side)

### DISCUSSION THREAD

**Discussion by vnea** [vercel/next.js#18924] [Help]
Hello,

On the client side, I need to use an environnement variable prefixed by `NEXT_PUBLIC_` in a **.env.local** file.
So I created a **.env.local** file with this content : 
```
NEXT_PUBLIC_TEST=BURGER
```

After that, I executed the following commands : 
```
next build
# Then I got this message
Loaded env from path/to/project/.env.local
info  - Creating an optimized production build 
...
Done in XX.YYs.
```
```
next start
# Then I got this message
Loaded env from path/to/project/.env.local
ready - started server on http://localhost:3000
```
After that, everything was OK and the value of `console.log(NEXT_PUBLIC_TEST)` was `BURGER` as expected.

But then, I tried to change the value of `NEXT_PUBLIC_TEST` to `BANANA` and I re-executed this command :
```
next start
# Then I got this message
Loaded env from path/to/project/.env.local
ready - started server on http://localhost:3000
```
But the content of  `console.log(NEXT_PUBLIC_TEST)` was still `BURGER` and not `BANANA`.
I had to re-execute `next build` and `next start` to update the value of `NEXT_PUBLIC_TEST`.

So it seems like environnement variables on **client side** can only be defined at **build time**.

**Question : Is there any way to define environnement variables on client side at run time instead of build rime ?**

I am using **next v9.5.5**.

**Comment by kachkaev** [comment]
You’ll need to use [runtime config](https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration) for that. Env vars are replaced with values on build time.

**Reply by vnea** [reply]
I tried this but it didn't work.
